Testing the format on pull request functionality

BlakeTNC opened the pull request:
testing format functionality

WiseEagleOwl commented:
@BlakeTNC I think the difference here is that your branch is part of this repository and not a remote fork, as it is for all other users. You are the exception to the rule as I understand it.
What I don't understand is how you achieved a branch with bad formatting. For me the formatting is already fixed for every push in my fork, so I had to remove the `on: push` part in #161. Sadly no format action is run at all in my PR which has me confused 😕

BlakeTNC commented:
@WiseEagleOwl 
You are correct. I was able to replicate your described issue. In case you are curious, I'll past my testing notes below. 
To resolve this problem, I've removed the "pull request" trigger from the action. Now, it will only run on on a "push" trigger. 
Let me know if this solves all problems. Let me know if there are any further problems in the future. 

Here were my testing steps, in case it is informative:
- I used a separate github account to create the fork. Account was "decenthappiness".
- Note, "LGDP" = LGoodDatePicker.
- with github web, created a new fork. (LGoodDatePicker/LGDP -> decenthappiness/LGDP)
- installed github desktop
- with github desktop, cloned the fork.
- set github desktop defaults to "contribute to parent repo"
- with a text editor, changed a file on local hard drive. (Changed code and formatting).
- with github desktop, committed locally.
- with github desktop, pushed to origin (pushed to decenthappiness/LGDP).
- I noted the format action did -not- run in the fork. This is expected. (Github disables all actions by default in forks.)
- with github web, created pull request from fork to master (to LGoodDatePicker/LGDP)
- The action FAILED: Format / formatting (pull_request)
- with github web and my own account, I modified the action to run only on push. (at LGoodDatePicker/LGDP)
- with github web and test account, created another pull request from fork to master.
- No formatting related action was run inside the pull request. All other actions succeeded. (aka, all actions succeeded.)
- with github web, I accepted and merged the pull request. 
- After the PR merge, the push format action ran automatically, as expected. The format action succeeded.

BlakeTNC commented:
PS, it's a mystery.
> What I don't understand is how you achieved a branch with bad formatting.

The file change and commit in this pull request #160 was initiated from the github web interface inside the (LGoodDatePicker/LGoodDatePicker) repository. Just guessing, using the web interface may have made a difference in how it was processed. 

However, the behavior of the 'pull_request' trigger does appear inconsistent. As you said, possibly depending on how the pull was initiated. For example, last night @dbmalkovsky did a pull request under #162. I don't know if his pull was initiated from a fork or not. In that pull request, the format action ran and succeeded. Looking at the log, I can confirm that successful format was triggered by the 'pull_request' trigger.

Anyway, as mentioned I removed the 'pull_request' trigger from the action. Now it is only push triggered. I think that is similar to how you fixed these problems before.
